extendr-api/src/wrapper/environment.rs

```diff
@@ -65,7 +65,9 @@ impl Environment {
             let robj = call!("new.env", TRUE, parent, dict_len).unwrap();
             for nv in names_and_values {
                 let (n, v) = nv.sym_pair();
-                unsafe { Rf_defineVar(n.get(), v.get(), robj.get()) }
+                if let Some(n) = n {
+                    unsafe { Rf_defineVar(n.get(), v.get(), robj.get()) }
+                }
```

Comment:
One thing not very obvious is what we should do when the user try to create an environment from a pairlist containing unnamed element. Currently it's just ignored. Should we raise an error?

Reply:
What is ignored? The unnamed element only, or the entire call? And if you triggered an error, would it be compile time or runtime?

Reply:
> The unnamed element only, or the entire call?

Currently this ignores the unnamed elements only.

> And if you triggered an error, would it be compile time or runtime?

Runtime.

Reply:
It's unclear to me. Usually, my philosophy is that code should use meaningful fallback behavior on empty input, so that would argue against raising an error. I have encountered so many cases where code would complain about `NA` or `NULL` input instead of silently ignoring it, and then I have to special-case these situations, and the code I'm calling is also special-casing, and it just doesn't make sense.

I assume the counter-argument is that silently ignoring invalid input can create difficult-to-find bugs.

Reply:
I'm getting inclined to raise an error, rather than handling the invalid inputs gracefully. I think this operation is for special cases when there's a lot of inputs generated programmatically, as the example shows. For more dynamic inputs, users can process the elements one by one with `Environment::set_local()` by themselves.

``` rust
    ///     let names_and_values = (0..100).map(|i| (format!("n{}", i), i));
    ///     let mut env = Environment::from_pairs(global_env(), names_and_values);
    ///     assert_eq!(env.len(), 100);
```

Reply:
Ok, sounds good to me.